### PR TITLE
Migrate from Poetry to uv

### PR DESCRIPTION
**Pull request by charliermarsh**

### CODE REVIEW

**Comment by charliermarsh** [comment]
Failing as expected due to absolute paths for now.

File: uv.lock [hunk 1]
```diff
@@ -0,0 +1,1195 @@
...
+name = "typing-extensions"
+version = "4.12.2"
+source = "registry+https://pypi.org/simple"
+marker = "python_version < '3.11'"
+
+[[distribution]]
+name = "autobot-ml"
+version = "0.0.16"
+source = "editable+file:///Users/crmarsh/workspace/autobot"
+sdist = { url = "file:///Users/crmarsh/workspace/autobot" }
```
Comment (T-256) [inline uv.lock:50]: it exposed local path of project, which is not good for whom respect privacy and also uploads their `uv.lock` publicly.

Comment (charliermarsh) [reply uv.lock:50]: Correct, we're changing it this week.

Comment (charliermarsh) [reply uv.lock:50]: See, e.g., https://github.com/astral-sh/uv/pull/4205.

**Pull request merged**